docs/getting_started/setup_your_environment.md

```diff
@@ -144,6 +144,30 @@ project (`npm init -y` as necessary), then add the following block to your
 }
 ```
 
+#### LSP clients
+
+Deno has builtin support for the
+[Language server protocol](https://langserver.org).
+
+If your editor supports the LSP, you can use Deno as a language server for
+TypeScript and JavaScript.
+
+The editor can start the server with `deno lsp`.
+
```

Comment:
It would be great if we could include example for other popular editor (vim/sublime). CC @uki00a I believe you set up lsp with vim?

Reply:
Yeah, I set up `deno lsp` with [`vim-lsp`](https://github.com/prabirshrestha/vim-lsp) like this:

```markdown
##### Example for Vim/Neovim

After installing the [`vim-lsp`](https://github.com/prabirshrestha/vim-lsp) LSP client
you can add the Deno language server by adding the following to your
`vimrc`/`init.vim`:

```vim
if executable("deno")
  augroup LspTypeScript
    autocmd!
    autocmd User lsp_setup call lsp#register_server({
    \ "name": "deno lsp",
    \ "cmd": {server_info -> ["deno", "lsp"]},
    \ "root_uri": {server_info->lsp#utils#path_to_uri(lsp#utils#find_nearest_parent_file_directory(lsp#utils#get_buffer_path(), "tsconfig.json"))},
    \ "whitelist": ["typescript", "typescript.tsx"],
    \ })
  augroup END
endif
```
```

Reply:
Was just about to write a PR to add this to the docs actually so great if we can include this here (also use vim-lsp).

Altho I wouldn't use tsconfig as the root, because most Deno projects don't have one.